node/node.go

```diff
@@ -114,6 +116,47 @@ func NewNode(reg *prometheus.Registry, config *Config, pubIPProvider pubip.Provi
 	// Create ChainState Client
 	cst := eth.NewChainState(tx, client)
 
+	var keyPair *core.KeyPair
+	var blsClient blssignerV1.SignerClient
```

Comment:
I feel like this one-of pattern where one is expected to be nil and the other is not is very brittle. 
There is this pattern in `Config` where `PrivateBls` can be empty and here where one of either `keyPair` or `blsClient` is nil. 
Downstream, these attributes can't be assumed to be non-nil and there needs to be if/else nil check whenever they're accessed. 
Is there a blsClient abstraction that allows local signing with private bls key?

Reply:
Agreed! This is what I was trying to describe in https://github.com/Layr-Labs/eigenda/pull/745#discussion_r1838528665

Reply:
> Is there a blsClient abstraction that allows local signing with private bls key?
I am thinking of having this abstraction in eigensdk and then you just initialize signer and based on config it would sign. Once I do that - we can switch it out.

Reply:
Is this something that will be done & reflected in this PR? Or in a future PR?